torch/_dynamo/variables/builder.py

```diff
@@ -461,6 +462,9 @@ def index_source(key):
                 source=self.source,
                 guards=make_guards(GuardBuilder.BUILTIN_MATCH),
             )
+        elif is_utils_checkpoint(value):
```

Comment:
This is basically the bugfix mentioned in the title. Other bits are just refactoring.

Reply:
Oh I see, it doesn't always construct the TorchHigherOrderOperatorVariable